tests/mir_test_framework/async_server_runner.cpp

```diff
@@ -79,18 +77,19 @@ mtf::AsyncServerRunner::AsyncServerRunner()
                 }
                 else
                 {
-                    result = std::make_shared<ml::FileLogger>(std::move(out));
+                    return std::make_shared<ml::FileLogger>(std::move(out));
                 }
             }
             else
             {
                 std::cerr << "Failed to open log file: " << output_filename << std::endl;
                 if (!server.get_options()->get<bool>(mtd::logging_opt))
                 {
-                  result = std::make_shared<mtd::NullLogger>();
+                    return std::make_shared<mtd::NullLogger>();
                 }
             }
-            return result;
+
+            return nullptr;
```

Comment:
I don't think we should ever return a null pointer from `the_logger()`

```suggestion
            return std::make_shared<ml::DumbConsoleLogger>();
```